node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -96,19 +96,19 @@ case class DataMessageHandler(
             }
           }
           newChainApi <- chainApi.processFilter(filter)
-        } yield {
-
-          Try {
-            val blockFilter =
-              BlockFilter.fromBytes(filter.filterBytes, filter.blockHash)
-            callbacks.onCompactFilterReceived.foreach(
-              _.apply(filter.blockHash, blockFilter))
-          } match {
-            case Failure(ex) =>
-              logger.error("Error processing compact filter", ex)
-            case Success(_) => ()
-          }
+          blockFilter <- Future(
+            BlockFilter.fromBytes(filter.filterBytes, filter.blockHash))
+          _ <- callbacks.onCompactFilterReceived
```

Comment:
So instead of defining this everywhere in `DataMessageHandler`, can we just create a helper method in `NodeCallbacks` that does this, and here we just call `_ <- NodeCallbacks.execute(callbacks)` or something like that?